api/tests/unit/util/test_logging.py

```diff
@@ -1,28 +1,70 @@
 import json
 import logging
+import os
+
+import pytest
 
 from util.logging import JsonFormatter
 
 
-def test_json_formatter__outputs_expected():
+@pytest.fixture
+def inspecting_handler() -> logging.Handler:
+    class InspectingHandler(logging.Handler):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self.messages = []
+
+        def handle(self, record):
+            self.messages.append(self.format(record))
+
+    return InspectingHandler()
+
+
+@pytest.mark.freeze_time("2023-12-08T06:05:47.320000+00:00")
+def test_json_formatter__outputs_expected(inspecting_handler: logging.Handler) -> None:
+    # Given
     json_formatter = JsonFormatter()
 
-    log_record = logging.LogRecord(
-        name="test_logger",
-        level=logging.INFO,
-        pathname="test.py",
-        lineno=42,
-        msg="This is a test message with args: %s and %s",
-        args=("arg1", "arg2"),
-        exc_info=None,
+    inspecting_handler.setFormatter(json_formatter)
+    logger = logging.getLogger("test_json_formatter__outputs_expected")
+    logger.addHandler(inspecting_handler)
+    logger.setLevel(logging.INFO)
+
+    expected_pid = os.getpid()
+    expected_tb_string = (
+        "Traceback (most recent call last):\n"
+        '  File "/Users/kgustyr/dev/flagsmith/flagsmith/api/tests/unit/util/test_logging.py",'
```

Comment:
Looks like you're going to need to fix the `$HOME` and work part of the filepath.

Reply:
Thanks! Done.